src/hoc/withTheme.js

```diff
@@ -31,18 +32,21 @@ const wrapWithTheme = (Component: ReactClass<any>) => {
     unsubscribeId: number = -1
 
     componentWillMount() {
+      const { defaultProps } = this.constructor
       const styledContext = this.context[CHANNEL_NEXT]
-      if (styledContext === undefined) {
+      const defaultTheme = determineTheme(this.props, undefined, defaultProps)
+      if (!styledContext && !defaultTheme) {
```

Comment:
Can we add `process.env.NODE_ENV !== "production"` here?

cc @mxstbr I don't think that such a message would be valuable outside of prod. This behaviour is so deterministic, that someone not passing anything might be an actual, real use-case sometimes.